src/notebook/plugin.ts

```diff
@@ -143,19 +172,26 @@ class NotebookFileHandler extends AbstractFileHandler {
     let model = new NotebookModel();
     let panel = new Panel();
 
-    let button = new Widget();
-    let b = document.createElement('button');
-    b.appendChild(document.createTextNode('Execute Current Cell'))
-    button.node.appendChild(b);
-
-
     let widgetarea = new Widget();
     let manager = new WidgetManager(widgetarea.node);
 
+    let notebookId = this.notebookId;
+    this.notebookId += 1;
+
     this.session.startNew({notebookPath: contents.path}).then(s => {
-      b.addEventListener('click', ev=> {
-        executeSelectedCell(model, s);
-      })
+      // TODO: it's probably better to make *one* shortcut that executes whatever
+      // the current notebook's selected cell is, rather than registering a
+      // a new shortcut for every open notebook.  To do this, we need the application
+      // state (or somewhere) to include the 'current document'
+      // this is better because it makes one shortcut, and it also is better because
+      // then it is easy for the command to be invoked via the menu, palette, plugin, etc.
```

Comment:
So you'd have to check if the current element is a document, and if not, if the previous element was a document.